File: code-coaches/medium/youtube-link-finder/youtube_link_finder.rb
def extract_video_id(url)
  if url.include?("youtube.com/watch")
      video_id = url.split("v=").last
  elsif url.include?("youtu.be/")
      video_id = url.split("/").last
  else
      raise ArgumentError, "Invalid YouTube video URL"
  end

  video_id
end

def main
  url = gets.chomp
  puts extract_video_id(url)
end

if __FILE__ == $0
  main
end

File: code-coaches/medium/youtube-link-finder/youtube_link_finder.py
def extract_video_id(url: str) -> str:
    if "youtube.com/watch" in url:
        video_id = url.split("v=")[1]
    elif "youtu.be/" in url:
        video_id = url.split("/")[-1]
    else:
        raise ValueError("Invalid YouTube video URL")

    return video_id


def main():
    url = input()
    print(extract_video_id(url))


if __name__ == "__main__":
    main()
